src/server.rs

```diff
@@ -866,6 +957,23 @@ fn main() {
                             }
                         }
                     }
+
+                    server_state.remove_timed_out_clients();
+
+                    // Send a KeepAlive every scond
+                    if (server_state.tick % 100) == 0 {
+                        for player in server_state.players.values() {
+                            let keep_alive = Packet::Response {
+                                sequence: player.next_resp_seq-1,
+                                request_ack: player.request_ack,
+                                code: ResponseCode::KeepAlive
+                            };
+
+                            tx.unbounded_send( (player.addr, keep_alive) ).unwrap();
+                        }
+                    }
```

Comment:
Ah, ok, so both sides are sending KeepAlive to the other, rather than the client sending them and the server replying to them. That works.